frigate/util.py

```diff
@@ -738,12 +738,59 @@ def escape_special_characters(path: str) -> str:
         return path
 
 
+def get_docker_memlimit_bytes() -> int:
+    """Get mem limit in bytes set in docker if present. Returns -1 if no limit detected"""
+
+    # prod container uses this path
+    memlimit_command = ["cat", "/sys/fs/cgroup/memory.max"]
+
+    p = sp.run(
+        memlimit_command,
+        encoding="ascii",
+        capture_output=True,
+    )
+
+    if p.returncode == 0:
+        value: str = p.stdout
+
+        if value.strip().isnumeric():
+            return int(value)
+        elif value.strip().lower() == "max":
+            return -1
+
+    # devcontainer seems to use this path
+    memlimit_command = ["cat", "/sys/fs/cgroup/memory/memory.max_usage_in_bytes"]
+
+    p = sp.run(
+        memlimit_command,
+        encoding="ascii",
+        capture_output=True,
+    )
+
+    if p.returncode != 0:
+        logger.debug(f"Unable to get docker memlimit: {p.stderr}")
+        return -1
+    else:
+        value: str = p.stdout
+
+        if value.strip().isnumeric():
+            return int(value)
+        elif value.strip().lower() == "max":
+            return -1
```

Comment:
I don't think it makes sense to handle this in the dev container. I don't expect our config to be setting memory limits and in any case I don't really see the benefit

Reply:
Well, apart from anything else it's far easier to test there. I still haven't got a "proper" docker build working locally, so had to get the path by attaching to a container and looking up the path manually.

I also tend to avoid different code in dev/release environments as far as possible to catch bugs earlier.

Reply:
The way I see it having code to specifically handle the dev container is different code. There's also already a number of differences in the way the dev container runs vs the release container as the dev container doesn't use S6, frigate is started manually, etc. so to me that is moot.

Just my 2 cents happy to hear what others think

Reply:
Can you give any pointers to producing a release docker image locally? I tried docker build . -t mybuild but it just produces another dev container from what I can see

Reply:
> Can you give any pointers to producing a release docker image locally? I tried docker build . -t mybuild but it just produces another dev container from what I can see

`make build`

Reply:
> > Can you give any pointers to producing a release docker image locally? I tried docker build . -t mybuild but it just produces another dev container from what I can see
> 
> `make build`

Thanks! I'm used to Java tooling in Eclipse/maven so VS Code, make, docker and devcontainers have been a bit of a learning curve